src/symbol/shaping.ts

```diff
@@ -590,6 +590,87 @@ function getAnchorAlignment(anchor: SymbolAnchor) {
     return {horizontalAlign, verticalAlign};
 }
 
+function calculateLineBlockMetrics(
+    glyphMap: {
+        [_: string]: {
+            [_: number]: StyleGlyph;
+        };
+    }, glyphPositions: {
+        [_: string]: {
+            [_: number]: GlyphPosition;
+        };
+    },
+    imagePositions: {[_: string]: ImagePosition},
+    line: TaggedString,
+    layoutTextSizeFactor: number
+) {
+    let maxGlyphHeight = 0;
+    let maxImageHeight = 0;
+    for (let i = 0; i < line.length(); i++) {
+        const section = line.getSection(i);
+
+        if (section.imageName) {
+            const imagePosition = imagePositions[section.imageName];
+            if (!imagePosition) continue;
+            maxImageHeight = Math.max(
+                maxImageHeight,
+                imagePosition.displaySize[1] * layoutTextSizeFactor
+            );
+        } else {
+            const codePoint = line.getCharCode(i);
```

Comment:
Something is wrong here. `i` iterator is used, but it stands for section number, while here it is used for char code. Probably another loop was intended.

Reply:
nvm, it is ok.

Reply:
Btw imho it should be `line.getSectionForChar(i)` because otherwise it looks like you are getting eg. third section, while actually you're getting section for 3rd character.